### DISCUSSION TITLE
FEATURE: Implementation in Rust...

### DISCUSSION THREAD

**Discussion by ZeroDot1** [funilrys/PyFunceble#306] [Ideas]
I recommend a complete new implementation of the program in Rust. 
With Rust, the whole program would run far more efficiently and the power consumption would be lower, which can only be an advantage for us in the current energy crisis.

**Comment by funilrys** [comment]
I have an early version of PyFunceble in Rust on my servers ... It's ugly and doesn't come to the level of PyFunceble (as of today) as a lot of features and code I abstract in Python (thanks to PyPI) are missing in cargo ... So I have to implement all of those myself and push them separately into the cargo ... That literally means that this project is a worldwide niche. 

As I'm developing another project/platform for better test orchestration, a public implementation in Rust won't be available yet.

**Comment by ZeroDot1** [comment]
Ah OK, that sounds good, if you need someone to test in the future let me know, I'll definitely volunteer.

**Comment by spirillen** [comment]
> Ah OK, that sounds good, if you need someone to test in the future let me know, I'll definitely volunteer.

Second this

**Comment by funilrys** [comment]
@spirillen @ZeroDot1 I will let you know! Testers are always welcome! 

Here is the first piece you may test if you are into whitelisting: 
- [Tivilsta](https://github.com/funilrys/tivilsta) an alternative to [UHBW](https://github.com/Ultimate-Hosts-Blacklist/whitelist-tool). It's not complete yet but it implements the same internal specification.

**Reply by spirillen** [reply]
This little sed is 10000 times faster as it do the work in under a second, where ./tivilsta takes more than a minute.

This one can easily be adapted to a `for while` and accept regex

```bash
remove-lines() (
    # usage: remove-lines file-with-lines-to-remove file-to-remove-lines-in
    remove_lines="$1"
    all_lines="$2"
    tmp_file="$(mktemp)"
    # To make this code remove lines containing. temp remove -Fx from arg
    grep -Fxvf "$remove_lines" "$all_lines" >"$tmp_file"
    mv "$tmp_file" "$all_lines"
)
```

Test line

```rust
./tivilsta -w ~/Projects/bitbucket/hosts-file.net/output/ad_servers.txt/domains/INACTIVE/list -s ~/Projects/bitbucket/hosts-file.net/ad_servers.txt -o ~/Projects/bitbucket/hosts-file.net/ad_servers.txt
```
